### ISSUE TITLE
%1 is not a valid Win32 application

### ISSUE THREAD

**Issue by nanodan** [pyca/cryptography#3351]
Installed cryptography with `pip install` on a 64 bit windows system running 64 bit version of Python 2.7.  When I try to run the following:

`from cryptography.fernet import Fernet`

I get the following error:

```
from cryptography.hazmat.bindings._padding import lib
ImportError: DLL load failed: %1 is not a valid Win32 application.
```

I tried uninstalling and installing with the whl

`cryptography-1.7.1-cp27-cp27m-win_amd64`

Same error.

**Comment by reaperhulk** [comment]
Hmm, I am completely unable to replicate this. If I install Python 2.7.13 (64-bit), install virtualenv, make a new virtualenv, then `pip install cryptography` and test the import I have no error. Could you give me more details of your env? Do you see this error if you first install (or upgrade to the latest) virtualenv, install virtualenv itself, and then pip install cryptography?

**Comment by Ayrx** [comment]
How did you install your Python?

**Comment by nanodan** [comment]
I installed Python from [https://www.python.org/ftp/python/2.7.13/python-2.7.13.amd64.msi](https://www.python.org/ftp/python/2.7.13/python-2.7.13.amd64.msi).  Tried uninstalling and reinstalling fresh, didn't help.

I tried to replicate this with a virtualenv on my home computer (also 64 bit windows w/ 64 bit python) and was unable to, the import line works fine.

So it seems like this issue is specific to my one work system and not the cryptography package.  I'll have to try and resolve that tomorrow and narrow down the problem.  Feel free to close this issue if you think its necessary, though I will follow up with resolution if found.

**Comment by schlenk** [comment]
When dynamically linking openssl such errors can happen if you have another libeay32.dll in your %WINDIR%\system32 folder (or Wow64 version for 32-Bit). Sadly that is quite common, as quite a few tools just drop an outdated libeay32.dll in a system folder.

But for the statically linked whl it makes no real sense. 

You can try Sysinternals procmon to see which dll (or more likely dependent DLL) the system is trying to load.

**Comment by reaperhulk** [comment]
Closing due to lack of activity

**Comment by unformatt** [comment]
Anyone find a solution to this?

**Comment by jairoxyz** [comment]
Yes a solution would be excellent. This still happens in a Python environment that finds other openssl libs in the PATH. In my case it works only in a clean virtual env. I also tried to use it with Kodi, which has its own Python embedded and here Kodi _ssl.pyd gets loaded before _padding.pyd, causing the same error. I understood that the static wheel would not load any system ssl libs but this is obviously not the case.

**Comment by reaperhulk** [comment]
@jairoxyz Could you open a new issue please describing the steps to reproduce? I think I understand, but I want to track it as a new issue so we can figure out what's going on.

**Comment by jairoxyz** [comment]
@reaperhulk thanks for answering so swiftly - am feeling a bit silly now because I can only reproduce this error when using cryptography packages built for x32 python on x64 python and vice versa. I have built cryptography  following your guide for building-cryptography-on-windows using Visual Studio Build Tools 2017 on Win Server 2012 with  vcvarsall.bat x86_amd64. The resulting package works in in Python 2.7.x 32bit on Windows10 x64. My dev env is Visual Studio with Python Tools.

I am still getting the error when using the packages on Kodi which is 32 arch:
Kodi compiled Dec 10 2017 by MSVC 190024215 for Windows NT x86 32-bit version 10.0 (0x0A000000)
Python 2.7.13 (default, Jul 13 2017, 20:56:54) [MSC v.1900 32 bit (Intel)]

I tried vcvarsall.bat amd64_x86 (this worked for pycryptodome) and vcvarsall.bat x86 but always get the the same error `%1 is not a valid Win32 application` so I assume I cannot build for NT x86 platforms? I also used pip ... no-binary :all: 

Shall I open an issue with steps how to reproduce this in Kodi?

Thanks in advance.
